src/app/swap/swap-listener.ts

```diff
@@ -0,0 +1,73 @@
+import { toSats } from "@domain/bitcoin"
+
+import { admin as LedgerAdmin } from "@services/ledger/admin"
+import { WalletCurrency } from "@domain/shared"
+import { LedgerTransactionType } from "@domain/ledger"
+import { SwapProvider, SwapState, SwapType } from "@domain/swap"
+import { addAttributesToCurrentSpan } from "@services/tracing"
+import { SwapErrorNoActiveLoopdNode } from "@domain/swap/errors"
+import { sha256 } from "@domain/bitcoin/lightning"
+
+export const startSwapMonitor = async (swapService: ISwapService) => {
+  const isSwapServerUp = await swapService.healthCheck()
+  if (isSwapServerUp instanceof Error === false) {
+    const listener = swapService.swapListener()
+    listener.on("data", (response) => {
+      // const swapData = response.parsedSwapData
+      // logger.info({ swapData }, "listenerSwapMonitor")
+      handleSwapOutCompleted(response)
+    })
+  } else {
+    return SwapErrorNoActiveLoopdNode
+  }
```

Comment:
the logic with === false is convoluted IMO
```suggestion
  if (isSwapServerUp instanceof Error) {
      return SwapErrorNoActiveLoopdNode
    })
  } else {
     const listener = swapService.swapListener()
    listener.on("data", (response) => {
      // const swapData = response.parsedSwapData
      // logger.info({ swapData }, "listenerSwapMonitor")
      handleSwapOutCompleted(response)

   }
```